test/Microsoft.AspNetCore.Antiforgery.Test/Internal/DefaultClaimUidExtractorTest.cs

```diff
@@ -111,5 +111,63 @@ public void DefaultUniqueClaimTypes_Present()
                 "nameIdentifierValue",
             }, uniqueIdentifierParameters);
         }
+
+        [Fact]
+        public void GetUniqueIdentifierParameters_PrefersSubClaimOverNameIdentifierAndUpn()
+        {
+            // Arrange
+            var identity = new ClaimsIdentity();
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, "nameIdentifierValue"));
+            identity.AddClaim(new Claim("sub", "subClaimValue"));
+            identity.AddClaim(new Claim(ClaimTypes.Upn, "upnClaimValue"));
+
+            // Act
+            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(identity);
+
+            // Assert
+            Assert.Equal(new string[]
+            {
+                "sub",
+                "subClaimValue"
+            }, uniqueIdentifierParameters);
+        }
+
+        [Fact]
+        public void GetUniqueIdentifierParameters_PrefersNameIdentifierOverUpn()
+        {
+            // Arrange
+            var identity = new ClaimsIdentity();
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, "nameIdentifierValue"));
+            identity.AddClaim(new Claim(ClaimTypes.Upn, "upnClaimValue"));
+
+            // Act
+            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(identity);
+
+            // Assert
+            Assert.Equal(new string[]
+            {
+                ClaimTypes.NameIdentifier,
+                "nameIdentifierValue"
+            }, uniqueIdentifierParameters);
+        }
+
+        [Fact]
+        public void GetUniqueIdentifierParameters_UsesUpnIfPresent()
+        {
+            // Arrange
+            var identity = new ClaimsIdentity();
+            identity.AddClaim(new Claim("fooClaim", "fooClaimValue"));
```

Comment:
Our tests use `foo` in thousands of places, what's wrong with it? It at least makes it clear that it's an arbitrary value...